protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -235,18 +232,93 @@ impl Behaviour {
 }
 
 impl NetworkBehaviour for Behaviour {
-    type ConnectionHandler = handler::Prototype;
+    type ConnectionHandler = Either<
+        handler::relayed::Handler,
+        Either<handler::direct::Handler, dummy::ConnectionHandler>,
+    >;
     type OutEvent = Event;
 
-    fn new_handler(&mut self) -> Self::ConnectionHandler {
-        handler::Prototype
+    fn handle_established_inbound_connection(
+        &mut self,
+        peer: PeerId,
+        connection_id: ConnectionId,
+        local_addr: &Multiaddr,
+        remote_addr: &Multiaddr,
+    ) -> Result<THandler<Self>, ConnectionDenied> {
+        let is_relayed = local_addr.iter().any(|p| p == Protocol::P2pCircuit); // TODO: Make this an extension on `Multiaddr`.
```

Comment:
Any thoughts on how to handle this? With the introduction of the new APIs, we are no longer passing a `ConnectedPoint` which has the `is_relayed` utility function on it.

Reply:
I am fine with:

- Adding a method `Multiaddr::is_relayed`
- Adding a method `MultiaddrExt::is_relayed`
- Keeping as is, i.e. `iter().any`